[nikele2001] iP

nikele2001 opened the pull request:
# DukePro
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://boulderbugle.com/quotes-185mla85))

DukePro frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ *SUPER* FAST to use

All you need to do is,
1. download it from here.
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practise Java too. Here's the `main` method:
```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

nikele2001 committed:
completed level 0

nikele2001 committed:
completed level 1

nikele2001 committed:
amended level 1 submission

nikele2001 committed:
completed level 2

nikele2001 committed:
completed level 3

nikele2001 committed:
completed level 4

nikele2001 committed:
completed level A-TextUiTesting, Level 5 and Level 6

nikele2001 committed:
completed A-Enums

nikele2001 committed:
added some documentation

nikele2001 committed:
added more documentation

nikele2001 committed:
fixed some bugs with regards to missing matcher.find() statements

nikele2001 committed:
implemented Level-7 on level 7 branch

nikele2001 committed:
completed Level-8 on level 8 branch

nikele2001 committed:
completed A-MoreOOP

nikele2001 committed:
Merge tag 'A-MoreOOP' into add-gradle-support

nikele2001 committed:
completed A-Gradle

nikele2001 committed:
completed A-JUnit

nikele2001 committed:
added more tests

nikele2001 committed:
Merge pull request #1 from nikele2001/A-JUnit

added more tests

nikele2001 committed:
completed A-Jar

nikele2001 committed:
completed A-JavaDoc

nikele2001 committed:
complete A-CodingStandard

nikele2001 committed:
complete Level-9

nikele2001 committed:
Merge tag 'A-JavaDoc'

nikele2001 committed:
merge A-JavaDoc and Level-9

nikele2001 committed:
fix bug regarding deletion of last task in tasklist

nikele2001 committed:
Merge branch 'A-CodingStandard'

nikele2001 committed:
fix a bug that does not process find commands correctly in parser

nikele2001 committed:
Update README.md

nikele2001 committed:
complete A-CheckStyle

nikele2001 committed:
Update README.md

nikele2001 committed:
complete Level-10

nikele2001 committed:
Merge branch 'master' of https://github.com/nikele2001/ip

nikele2001 committed:
Fix some styling errors

LimJH2002 commented on README.md:
I like how you add in your own modifications to the README file 😄

LimJH2002 commented on src/main/java/duke/DialogBox.java:
```diff
@@ -0,0 +1,61 @@
+import javafx.scene.control.Label;
+import javafx.scene.image.Image;
+import javafx.scene.image.ImageView;
+import javafx.scene.layout.HBox;
```
Great job in splittting the imports to make it cleaner!

LimJH2002 commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,169 @@
+ * A class that handles all the user interactions of the program
+ */
+public class Ui {
+    private String lnspace = "____________________________________________________________";
```
You could also try making this final and changing the variable name to LNSPACE. But this works too!

LimJH2002 commented on src/main/java/duke/command/Command.java:
Awesome job in splitting your Command classes!

LimJH2002 commented on src/test/java/duke/TaskListTest.java:
```diff
@@ -0,0 +1,33 @@
+
+public class TaskListTest {
+    @Test
+    public void emptyTaskListLengthTest() {
```
Great names!

nikele2001 committed:
Implement code quality

load() in Storage.java: method too long

The load() method is too long and it exceeds 30 lines of code.

Abstracting out repetitive parts of the method makes it more readable for others working on the same code base.

Defining a method to add stored tasks, addStoredTask(), shortens the load() method and makes it more understandable.

nikele2001 committed:
Add assert statements into Parser

getIndex() method: asserted that there should be only 2 elements in the string array after splitting the user command.

This is to ensure that there are no subsequent information that user has inputted to ensure the integrity of what the user is trying to do.

nikele2001 committed:
Merge pull request #2 from nikele2001/A-Assertions

Add assert statements into Parser

nikele2001 committed:
Create ViewSchedule Command

View command: returns all tasks recorded that fulfils the 2 criterias:
- The task is not marked.
- Depending on the task type, it is considered relevant.
- For Todo tasks, all tasks are considered relevant.
- For Deadline tasks, tasks are considered relevant if deadline date has not passed.
- For Event tasks, tasks are considered relevant if provided date falls within Event duration.

The View command provides a way to view tasks in the form of a schedule e.g., view the schedule for a specific date.

To use the command:
enter "view " followed by a date in the form of a formatted string in the given format: "yyyy-mm-dd".

nikele2001 committed:
Merge pull request #3 from nikele2001/branch-BCD-Extension

Create ViewSchedule Command

nikele2001 committed:
fix bugs in view command, create Ui.png and create user guide

nikele2001 committed:
fix jar file generation

nikele2001 committed:
fixed some stuff
